iconbutton/lib/_standard-icon-button.scss

```diff
@@ -35,8 +35,10 @@
       --_#{$token}: var(--md-icon-button-#{$token}, #{$value});
     }
 
-    height: var(--_state-layer-size);
-    width: var(--_state-layer-size);
+    max-height: var(--_state-layer-size);
+    max-width: var(--_state-layer-size);
+    width: 100%;
```

Comment:
Icon buttons should not expand to fill the width/height of their container, so I think using width/height for the state layer size is correct.

Can you give some examples where max-* is needed? #4559 looks like it might be flexbox-related

Reply:
@asyncLiz 
It's a Mozilla and Apple thing, if you don't set min or max then they will ignore height or width.

Reply:
Iow setting min/max and height/width to 100% does what we expect width/height does on chrome

Reply:
https://lit.dev/playground/#gist=7c7c9aa6d2a05b5f273d60df84c8a91e

Reply:
https://lit.dev/playground/#gist=7c7c9aa6d2a05b5f273d60df84c8a91e

Reply:
This is what I'd expect with a `width: 100%` item in a flex container with other siblings.

You can fix this by setting `flex: 1` on the filler instead of `width: 100%`. That tells flex box to fill the remaining space, while not trying to squish other siblings to try and make the filler truly "100%" the width of the container.

Alternatively, you can set `flex-shrink: 0` on specific items to tell the flex box not to squish them.

Reply:
@asyncLiz OMG, my bad! thanks for checking it out!,`flex-shrink` cool learned something new!